app.py

```diff
@@ -401,9 +366,29 @@ def backburner_populate_shotgun(self, info, comments):
         self.log_debug("Created a version in Shotgun: %s" % sg_version_data)
         
         # upload quicktime to Shotgun
-        self.log_debug("Begin upload of quicktime to Shotgun...")
-        self.shotgun.upload("Version", sg_version_data["id"], full_path, "sg_uploaded_movie")
-        self.log_debug("Upload complete!")
+        if self.get_setting("bypass_shotgun_transcoding"):
+            self.log_debug("Begin upload of explicit mp4 quicktime to Shotgun...")
+            self.shotgun.upload(
+                "Version",
+                sg_version_data["id"],
+                full_path,
+                "sg_uploaded_movie_mp4"
+            )
+            self.log_debug("Upload complete!")
+
+            # upload thumb
+            self._upload_thumbnail(
+                full_path,
+                info["width"],
+                info["height"],
+                "Version",
```

Comment:
yes - there is a `share_thumbnail` method in the sg API but I thought it was easier to keep the code simple for now. This runs as a renderfarm task so it's wont really affect performance that much.